packages/server/src/modules/auth/auth.resolver.ts

```diff
@@ -180,20 +182,72 @@ export class AuthResolver {
     });
   }
 
-  @Mutation(() => GraphQLVoid, { nullable: true })
-  async sendRegistrationInvite(
+  @Mutation(() => GraphQLUser, { nullable: true })
+  async addOrInviteOperator(
     @Input() { email, retailLocationId }: RegistrationInviteLinkPayload,
-    @CurrentUser() operator: User,
+    @CurrentUser() currentUser: User,
   ) {
+    await this.authService.assertMembership({
+      userId: currentUser.id,
+      message: "You are not allowed to add a new Operator",
+      role: "ADMIN",
+      retailLocationId,
+    });
+
+    const userToUpdate = await this.userService.findUserByEmail(email);
+
+    if (userToUpdate) {
```

Comment:
Invert the conditional branch. Rename `userToUpdate` to `existingUser`. So, instead of doing `if (existingUser) { longUpdateLogic; return; } shortInviteLogic;`, do `if (!existingUser) { shortInviteLogic; } longUpdateLogic;`